src/ui/public/state_management/state_storage/state_hash.js

```diff
@@ -0,0 +1,29 @@
+import { Sha256 } from 'ui/crypto';
+
+// This prefix is used to identify hash strings that have been encoded in the URL.
+const HASH_PREFIX = 'h@';
+
+export function createStateHash(json, existingJsonProvider) {
+  if (typeof json !== 'string') {
+    throw new Error('createHash only accepts strings (JSON).');
+  }
```

Comment:
The assertion is really only for the purpose of helping developers know when they're using the function wrong, I think it's okay to leave